pkg/booleanpolicy/evaluator/mapeval/matcher.go

```diff
@@ -20,8 +20,8 @@ const (
 )
 
 type kvConstraint struct {
-	key   regexutils.WholeStringMatcher
-	value regexutils.WholeStringMatcher
+	key   regexutils.StringMatcher
```

Comment:
Curious unless the answer is below.  Why does the key not remain a `WholeStringMatcher`?